src/anomalib/data/__init__.py

```diff
@@ -144,6 +145,28 @@ def get_datamodule(config: DictConfig | ListConfig) -> AnomalibDataModule:
             val_split_mode=config.dataset.val_split_mode,
             val_split_ratio=config.dataset.val_split_ratio,
         )
+    elif config.dataset.format.lower() == "csv":
+        datamodule = CSV(
+            root=config.dataset.root,
+            normal_csv=config.dataset.normal_csv,
+            abnormal_csv=config.dataset.abnormal_csv,
+            task=config.dataset.task,
+            normal_test_csv=config.dataset.normal_test_csv,
+            mask_csv=config.dataset.mask_csv,
```

Comment:
I was wondering if it were possible to pass a single csv file rather than multiple such as `normal_csv`, `abnormal_csv`, `normal_test_csv` and `mask_csv`.

This csv file would store all these information in a single column. Overall it would be something like the following:

```
    |---|---------------|-------|---------|---------------|---------------------------------------|-------------|
    |   | path          | split | label   | image_path    | mask_path                             | label_index |
    |---|---------------|-------|---------|---------------|---------------------------------------|-------------|
    | 0 | datasets/name |  test |  defect |  filename.png | ground_truth/defect/filename_mask.png | 1           |
    |---|---------------|-------|---------|---------------|---------------------------------------|-------------|
```

of course some of these columns would be optional. depending on their availability, the details of the task and image/mask filenames would be created.

What do you think?

Reply:
That's actually a great idea! 

To clarify your thinking on some of the columns, especially the difference between `path` and `image_path`? Factoring in the `root` in config:
- `path`: Same function as `root` - absolute root path where the images are located
- `image_path`: Relative path to the image. Absolute path will be a function of `path / image_path`

A few more conventions:
- `split` column: possible values `train`, `test` and `val`
- `label` columns: `normal` and `defect`, or `normal` and `abnormal` - which one do you think aligns more with the terminologies used through the anomalib library?

Reply:
yeah, you are right. `path` would be the `root`. Maybe we may not even always need `path/image_path` if `image_path` is an absolute path. 

regarding the `label`, `normal` vs `abnormal` or `anomalous` would be better than `defect`

Reply:
Agreed. Will include checking `is_absolute()` for `image_path` if we need to construct `root/image_path` just so we can handle both types if users so choose to define.

Noted on the labels, I thought so too!